src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -1845,23 +1849,36 @@ private void LogConflict(Reference reference, string fusionName)
         /// <summary>
         /// Reads the state file (if present) into the cache.
         /// </summary>
-        private void ReadStateFile()
+        private void ReadStateFile(GetLastWriteTime getLastWriteTime, AssemblyTableInfo[] installedAssemblyTableInfo)
```

Comment:
Why this change?

Reply:
I know you said GetLastWriteTime was only provided for testing, but this is in case that ever changes. I wanted to update the `LastModified` field of each fileState to what it would be if it's queried later, and since it technically uses getLastWriteTime, I wanted to use the appropriate getLastWriteTime, which meant passing it through to DeserializePrecomputedCaches and using it to adjust LastModified rather than using the real timestamp.

Reply:
I added installedAssemblyTableInfo because they were set together, and it seemed wrong to split them. That part wasn't actually needed, though.